### DISCUSSION TITLE
Help with Simple RLS Policy

### DISCUSSION THREAD

**Discussion by everettglovier** [supabase/supabase#28149] [Questions] [answered]
I have a projects table with **id** and **user_id**. This table has a simple policy:
```
(auth.uid() = user_id)
``` 
and works totally fine for inserting and selecting a project.

I have another table called **characters** with **project_id**. The policy is this:
```
 (project_id IN ( SELECT projects.id FROM projects))
```

My understanding is that the RLS policy for **projects** will apply to the **characters** SELECT and return all projects belonging to the authed user. But, it fails. If i disable RLS on **projects**, it succeeds. What am I doing wrong?

**Comment by GaryAustin1** [comment]
Have you tested it in the SQL editor with impersonation to rule out your call somehow not having an authorized user?
RLS should limit the rows as you suggest.  It would still be better to add a Where clause on user_id = auth.uid() to it so you are not relying on RLS of other tables and for performance.

**Reply by everettglovier** [reply]
So to test, if I turn off RLS on the projects table, I can INSERT into the characters table, no problem. I've also tried expanding the query to include the where clause you mentioned, but same issue. This almost feels like a bug of some sort?

**Reply by GaryAustin1** [reply, marked as answer]
Did you test it from the SQL editor?
Not likely a bug as this is all Postgres stuff.
If you did not test from the SQL editor then you need to verify you have a user session with getSession before the call.

Also you will probably need to show your policies on both tables including their TO clause and the type of policy (permissive/restrictive).

**Reply by everettglovier** [reply]
First of all, thank you for taking the time to help me. I never knew about impersonation and you just blew my mind and saved me tons of time!

Everything worked fine when impersonating in the SQL editor. I think my issue must be client side.
